### PR TITLE
refactor(api-markdown-documenter): Unify configuration patterns

### PR DESCRIPTION
**Pull request by Josmithr**
This library has an inconsistent mix of `Partial` and `Required` types to represent partial user input parameters and "complete" configurations needed by the system to function.

This PR begins an attempt to align its APIs with the following conventions:

-   Naming:
    -   "Options": refers to user-provided API parameters, which may be incomplete.
    -   "Configuration": refers to the "complete" sets of parameters needed by system functionality.
-   Typing:
    -   When possible, "configuration" types will be declared with all properties required.
    -   When possible, "options" types will be declared as `Partial<FooConfiguration>`. When not possible, they will be declared as separate types.

Specifically, this PR replaces the `ApiItemTransformationConfiguration` type with `ApiItemTransformationOptions` and `ApiItemTransformationConfiguration`.

It also renames the existing `ApiItemTransformationOptions` type to `ApiItemTransformations`, which better aligns it with similar concepts in the renderers.

### CODE REVIEW

File: tools/api-markdown-documenter/CHANGELOG.md [hunk 1]
```diff
@@ -8,24 +8,6 @@
...
--   `ApiItemTransformationConfiguration`
--   `ApiItemTransformationOptions`
--   `DocumentationSuiteOptions`
--   `HtmlRenderer.RenderHtmlConfig`
--   `LintApiModelConfiguration`
--   `MarkdownRenderer.Renderers`
--   `MarkdownRenderer.RenderContext`
--   `ToHtmlTransformations`
-
 ##### Example
```
Comment (Josmithr) [inline tools/api-markdown-documenter/CHANGELOG.md:11]: This was supposed to be under the earlier heading, but was accidentally moved in a previous PR.

File: tools/api-markdown-documenter/src/api-item-transforms/configuration/Configuration.ts [hunk 2]
```diff
@@ -44,18 +43,40 @@ export interface ApiItemTransformationConfiguration
...
+ * @public
+ */
+export interface ApiItemTransformationConfiguration
+	extends ApiItemTransformationConfigurationBase,
+		ApiItemTransformations,
+		Required<DocumentationSuiteOptions>,
+		Required<LoggingConfiguration> {}
+
+/**
+ * API Item transformation configuration.
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/api-item-transforms/configuration/Configuration.ts:58]:
```suggestion
 * Input options for API Item transformation functionality.
```

File: tools/api-markdown-documenter/src/api-item-transforms/configuration/DocumentationSuiteOptions.ts [hunk 3]
```diff
@@ -91,6 +91,10 @@ export type HierarchyBoundaries = ApiMemberKind[];
 /**
  * Options for configuring the documentation suite generated by the API Item -\> Documentation Domain transformation.
  *
+ * @remarks
+ * Used as API input.
+ * Optional properties will be filled in with default values if not provided.
+ *
  * @public
  */
 export interface DocumentationSuiteOptions {
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/api-item-transforms/configuration/DocumentationSuiteOptions.ts:100]: TODO: didn't mean to bundle these changes here. Will revert them for now, and make the appropriate changes in a follow-up PR.

File: tools/api-markdown-documenter/src/api-item-transforms/configuration/Configuration.ts [hunk 4]
```diff
@@ -44,18 +43,44 @@ export interface ApiItemTransformationConfiguration
...
+ * Input options for API Item transformation APIs.
+ *
+ * @public
+ */
+export interface ApiItemTransformationOptions
+	extends ApiItemTransformationConfigurationBase,
+		Partial<ApiItemTransformations>,
+		DocumentationSuiteOptions,
+		LoggingConfiguration {}
+
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/api-item-transforms/configuration/Configuration.ts:71]: This is the crux of this PR. Explicit separate types. Allows the removal of _many_ usages of `Required<ApiItemTransformationConfiguration>`.

File: tools/api-markdown-documenter/src/api-item-transforms/configuration/Transformations.ts [hunk 5]
```diff
@@ -59,7 +59,7 @@ export type TransformApiItemWithoutChildren<TApiItem extends ApiItem> = (
  *
  * @public
  */
-export interface ApiItemTransformationOptions {
+export interface ApiItemTransformations {
```
Comment (Josmithr) [inline tools/api-markdown-documenter/src/api-item-transforms/configuration/Transformations.ts:62]: Rename required to allows creating separate `Configuration` and `Options` types. But it also aligns this type better with similar "transformations" types used elsewhere in the library (e.g., in the "renderers").

File: tools/api-markdown-documenter/src/api-item-transforms/configuration/index.ts [hunk 6]
```diff
@@ -16,8 +18,8 @@ export {
 	type HierarchyBoundaries,
 } from "./DocumentationSuiteOptions.js";
 export {
-	type ApiItemTransformationOptions,
-	getApiItemTransformationOptionsWithDefaults,
+	type ApiItemTransformations,
+	getApiItemTransformationsWithDefaults as getApiItemTransformationOptionsWithDefaults,
```
Comment (alexvy86) [inline tools/api-markdown-documenter/src/api-item-transforms/configuration/index.ts:22]: Keeping the alias is intended?

File: tools/api-markdown-documenter/src/api-item-transforms/configuration/Configuration.ts [hunk 7]
```diff
@@ -13,19 +13,18 @@ import {
...
-} from "./TransformationOptions.js";
+	type ApiItemTransformations,
+	getApiItemTransformationsWithDefaults,
+} from "./Transformations.js";
 
 /**
- * API Item transformation configuration.
+ * Shared base type for {@link ApiItemTransformationConfiguration} and {@link ApiItemTransformationOptions}.
+ *
+ * @remarks Not intended to be used directly.
```
Comment (alexvy86) [inline tools/api-markdown-documenter/src/api-item-transforms/configuration/Configuration.ts:23]: `@system` + `@sealed`?

**Review by alexvy86 — APPROVED** [review]
🥳 one comment

File: tools/api-markdown-documenter/src/api-item-transforms/configuration/index.ts (hunk 6 above)
Comment (Josmithr) [reply tools/api-markdown-documenter/src/api-item-transforms/configuration/index.ts:22]: Ugh. Nope. Good catch. Thanks!

File: tools/api-markdown-documenter/src/api-item-transforms/configuration/Configuration.ts (hunk 7 above)
Comment (Josmithr) [reply tools/api-markdown-documenter/src/api-item-transforms/configuration/Configuration.ts:23]: This library doesn't support the `@system` tag - that's a concept specific to the client release group. It probably makes sense more generally, but the tooling won't support it here yet.

But `@sealed` is a good call. Will update.

File: tools/api-markdown-documenter/src/api-item-transforms/configuration/index.ts (hunk 6 above)
Comment (Josmithr) [reply tools/api-markdown-documenter/src/api-item-transforms/configuration/index.ts]: Actually, this didn't need to be exported at all! Removed!

**Pull request merged**